core-tests/src/test/java/org/mule/runtime/core/el/context/MessageAttachmentsTestCase.java

```diff
@@ -108,15 +108,15 @@ public void outboundAttachment() throws Exception
     {
         MuleEvent event = getTestEvent("");
         DataHandler dataHandler = mock(DataHandler.class);
-        event.getMessage().addOutboundAttachment("foo", dataHandler);
+        ((DefaultMuleMessage) event.getMessage()).addOutboundAttachment("foo", dataHandler);
```

Comment:
Why aren't you using MutableMuleMessage here? Seems like something that should be replaced with the builder but there's no reference to that.